src/app/charts/chart.component.ts

```diff
@@ -137,6 +158,11 @@ export class ChartComponent implements OnChanges {
         this.imageExportService.downloadAsPNG(this.chart.indicator.name, fileName);
     }
 
+    public onThresholdSelected($event) {
+        delete $event.event; // everything else is a threshold parameter
```

Comment:
Gotcha. Do you have a link to that approach? If it's standard convention, then I'd be more ok with it. 

Maybe we could split the difference and do something like:
```es6
public onThresholdSelected($event) {
    // arbitrarily chose data as a more generic name that isn't part of the Event spec
    const thresholdParams = $event.data as ThresholdIndicatorQueryOpts
    ...
}
```
Feels a bit more clear, and still gives easy access to both the data packaged with the event and the event itself. Thoughts?